src/Elmish.WPF/DynamicViewModel.fs

```diff
@@ -75,19 +75,25 @@ type [<AllowNullLiteral>] internal DynamicViewModel<'model, 'msg>
         let! vmBinding =
           Initialize(loggingArgs, binding.Name, getFunctionsForSubModelSelectedItem)
             .Recursive(initialModel, dispatch, (fun () -> currentModel), binding.Data)
-        do bindingDict.Add(binding.Name, vmBinding)
-        let! errorList = FirstValidationErrors().Recursive(vmBinding)
-        do validationDict.Add(binding.Name, errorList)
-        return ()
-      } |> Option.defaultValue ()
+        option {
```

Comment:
Nested optional CEs? I think the code in `master` is better and that this commit half undoes your previous few commits.

Reply:
So the code in `master` won't allow me to go where this is going. But I think I can revert the nesting and still get there.

Reply:
Ok I have this changed to `Option.iter` for the internal one

Reply:
Where is this going?

Reply:
In the next PR #524 we move out this initializer from inside the declaration of the binding dictionary, and change the declaration of bindings *and* validation errors to just be for bindings. Then in *the next* PR #523 this all gets extracted out into a helper, which in #522 gets reused in `BaseViewModel<'model, 'msg>`

Reply:
Importantly, the Helper in PR #523 has to manage the validation errors dictionary. This is because errors work differently from bindings in that they are all accessed from the `INotifyDataErrorInfo` interface instead of the getters and setters. Since the helper is doing all of the `INotifyDataErrorInfo` heavy lifting, it needs access to the `validationErrors` dictionary, while the `bindings` need to be managed by the getters and setters which aren't managed by the helper.

Additionally the helper's `UpdateModel` has to take in the current list of `bindings` as a parameter, which allows it to do the heaving lifting of the `INotifyPropertyChanged` interface even without storing the `bindings` in the helper like we store the `validationErrors`.

Just as an aside, we *could* use the `bindings` parameter of `UpdateModel` and run `FirstValidationErrors().recursive(each binding.Value) |> Option.iter (fun errorList -> helperErrors[binding.Key] <- errorList)` every time the model updates, but that seemed inefficient.

Reply:
Still think this code is worse, but I am going to trust you that this will be temporary.